app/models/sqs.go

```diff
@@ -222,6 +223,71 @@ func (r *SendMessageRequest) SetAttributesFromForm(values url.Values) {
 	}
 }
 
+func NewSendMessageBatchRequest() *SendMessageBatchRequest {
+	return &SendMessageBatchRequest{}
+}
+
+type SendMessageBatchRequest struct {
+	Entries  []SendMessageBatchRequestEntry
+	QueueUrl string
+}
+
+func (r *SendMessageBatchRequest) SetAttributesFromForm(values url.Values) {
+	for key := range values {
+
+		keySegments := strings.Split(key, ".")
+		if len(keySegments) <= 3 {
+			// no attribute
+			continue
+		}
+		entryIndex, err1 := strconv.Atoi(keySegments[1])
+		attributeIndex, err2 := strconv.Atoi(keySegments[3])
+		if err1 != nil || err2 != nil {
```

Comment:
The case the number of key segments will be 4 (index 0 to 3) like: `Entries.1.MessageAttributes.1.Name`
The case that the number will be under 3 like: `Entries.1.MessageBody`

Reply:
Ah I see.  Could you make a note of why the 2 are there?  There's no way I'm going to remember that the Message Attributes have extra layers to them.